### PR TITLE
feat: add search functionality with debounce

### PR DESCRIPTION
**Pull request by rajat-ws**
## Description:

- added search functionality with debounce
- gif added for preview

### CODE REVIEW

File: containers/Itunes/index.tsx [hunk 1]
```diff
@@ -14,10 +18,22 @@ const Container = styled.div`
 `;
 
 export const TracksContainer = () => {
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const { data } = useFetchTracksQuery(searchTerm, { skip: !searchTerm });
+  console.log(data);
```
Comment (praveenkumar1798) [inline containers/Itunes/index.tsx]: remove logs

File: containers/Itunes/index.tsx [hunk 2]
```diff
@@ -14,10 +18,22 @@ const Container = styled.div`
...
+
+  const handleOnChange = trackName => {
+    setSearchTerm(trackName);
+  };
```
Comment (praveenkumar1798) [inline containers/Itunes/index.tsx]:
```
const handleOnChange = trackName => setSearchTerm(trackName)
  ```

File: containers/Itunes/index.tsx [hunk 3]
```diff
@@ -14,10 +18,22 @@ const Container = styled.div`
...
+
+  const debouncedHandleOnChange = debounce(handleOnChange, 200);
+
+  // console.log("data", data);
```
Comment (praveenkumar1798) [inline containers/Itunes/index.tsx:28]: remove comments

**Review by praveenkumar1798 — APPROVED** [review]

File: containers/Itunes/index.tsx (hunk 1 above)
Comment (rajat-ws) [reply containers/Itunes/index.tsx]: Updated accordingly.

File: containers/Itunes/index.tsx (hunk 2 above)
Comment (rajat-ws) [reply containers/Itunes/index.tsx]: Updated accordingly.

**Pull request merged**